mlir/test/Gradient/AdjointTest.mlir

```diff
@@ -12,11 +12,12 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// RUN: quantum-opt %s --lower-gradients=only=adj --split-input-file --verify-diagnostics | FileCheck %s
+// RUN: quantum-opt %s --lower-gradients=only=adj --split-input-file --verify-diagnostics --allow-unregistered-dialect | FileCheck %s
```

Comment:
For what dialect it was necessary?

Reply:
This was causing an issue where a QNode being differentiated using the adjoint method must have a quantum register, but the quantum register in these tests was being canonicalized away by the `applyPatternsAndFoldGreedily` inside of the `--lower-gradients` pass because the quantum register was not used.

A workaround for this was to use an unregistered op to prevent the register from being canonicalized away.